### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by BonyBabu**
added missing version info for plugins maven-surefire-plugin and com.mycila.maven-license-plugin.
added Javadoc dependency colt in pom
updated the outdated JavaDoc parameter's documentation for some functions.

### CODE REVIEW

File: datex2/pom.xml [hunk 1]
```diff
@@ -96,6 +97,7 @@
 				<configuration>
 					<excludedGroups>broken</excludedGroups>
 				</configuration>
+                <version>${maven.surefire.plugin.version}</version>
```
Comment (stephencurran) [inline datex2/pom.xml]: The indentation looks wrong here.

File: decoder/pom.xml [hunk 2]
```diff
@@ -57,7 +57,8 @@
 
 			<plugin>
 				<groupId>com.mycila</groupId>
-				<artifactId>license-maven-plugin</artifactId>
+				<artifactId>maven-license-plugin</artifactId>
+                <version>${maven.license.plugin.version}</version>
```
Comment (stephencurran) [inline decoder/pom.xml]: The indentation looks wrong here.

File: pom.xml [hunk 3]
```diff
@@ -48,6 +48,8 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+         <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (stephencurran) [inline pom.xml]: The indentation looks wrong here.

File: datex2/pom.xml (hunk 1 above)
Comment (stephencurran) [inline datex2/pom.xml]: The indentation looks wrong here.

Comment (BonyBabu) [reply datex2/pom.xml]: 1fe5940d667da0881ad8595be524a396dc69a0e0

File: decoder/pom.xml (hunk 2 above)
Comment (BonyBabu) [reply decoder/pom.xml]: addressed in 1fe5940d667da0881ad8595be524a396dc69a0e0

File: pom.xml (hunk 3 above)
Comment (BonyBabu) [reply pom.xml]: addressed in 1fe5940d667da0881ad8595be524a396dc69a0e0

File: datex2/pom.xml (hunk 1 above)
Comment (BonyBabu) [reply datex2/pom.xml]: addressed in 1fe5940d667da0881ad8595be524a396dc69a0e0

**Pull request merged**